scico/solver.py

```diff
@@ -579,7 +578,7 @@ def golden(
     return r
 
 
-class ATADSolver:
+class MatrixATADSolver:
     r"""Solver for linear system involving a symmetric product plus a diagonal.
```

Comment:
Similar to above, maybe something like "Solver to invert A^* W A + D, where..."

Reply:
Same issue as above: it's not straightforward to actually fit this into a single line.